core/src/main/scala/spinal/core/internals/ComponentEmitterVhdl.scala

```diff
@@ -624,19 +677,21 @@ class ComponentEmitterVhdl(
         statement match {
           case assignment: AssignmentStatement  => b ++= emitAssignment(assignment,tab, assignmentKind)
           case assertStatement: AssertStatement =>
-            val cond = emitExpression(assertStatement.cond)
+            if (!spinalConfig.formalAsserts) {
+              val cond = emitExpression(assertStatement.cond)
 
-            require(assertStatement.message.isEmpty || (assertStatement.message.size == 1 && assertStatement.message.head.isInstanceOf[String]))
+              require(assertStatement.message.isEmpty || (assertStatement.message.size == 1 && assertStatement.message.head.isInstanceOf[String]))
```

Comment:
line 683 and 685 would better not be committed.

Reply:
What do you mean by that? The lines 683/685/687 were already existing, I simply wrapped them in the `if (!spinalConfig.formalAsserts)` conditional. They are perfectly valid, and even required, to generate the appropriate assert statements in an non-formal context (usually simulation) where report messages and severities are allowed.